.github/workflows/reusable-generate-mermaid-svg.yaml

```diff
@@ -0,0 +1,65 @@
+#########################################################################################
+# Copyright (c) 2023 Contributors to the Eclipse Foundation
+#
+# See the NOTICE file(s) distributed with this work for additional
+# information regarding copyright ownership.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache License, Version 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0.
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+#########################################################################################
+
+name : 'generate static Mermaid files'
+on:
+  workflow_call:
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: setup node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+      - name: install mermaid cli
+        run: |
+          npm install -g @mermaid-js/mermaid-cli
+      - name: print mmdc version
+        run: |
+          mmdc -V
+      - name: checkout source repo
+        uses: actions/checkout@v3.5.3
+      - name: search for files
+        id: search_files
+        run: |
+          pwd
+          echo "mermaid_files=$(find . -type f \( -name "*.mmd" -o -name "*.mermaid" \) | xargs )" >> $GITHUB_OUTPUT
+      - name: show step output
+        run: echo "step files from ${{ steps.search_files.outputs.mermaid_files }}"
+      - name: generate svg file
+        id: generated
+        run: |
+          for file in ${{ steps.search_files.outputs.mermaid_files }};
+            do
+             mmdc -i $file -o ${file%.*}.svg
+             echo "${file%.*}.svg"
+             git add ${file%.*}.svg
```

Comment:
I am unsure, what this is doing. `git add`, without commit? Is this needed, it there is an `git-auto-commit-action`
In general I still think, that it is not a good idea to autocommit things that where generated from a PR.
The minimal change, that I would want to see here, is an environment variable that controls, whether or not the auto-commit is done. It should default to `false`

Reply:
at this step it will only staged, in case of there are more files that have changes.
At the current point this workflow just keep the static files updated.

Reply:
``` mermaid
---
 title: workflow currently works
---
flowchart TD
    A[Reason: I want to change puml / mmd ] --> B(create PR)
    B --> C(someone approve pr on codelines)
    C --> D(merge pr)
    D --> E(Workflow checks that there are changed puml/mmd codefiles)
    E --> F(workflow provides new static files )
    F --> G(update static files on main)
```